pdns/dnsname.hh

```diff
@@ -63,6 +63,15 @@ class DNSName
 {
 public:
   DNSName()  {}          //!< Constructs an *empty* DNSName, NOT the root!
+  // Work around assertion in some boost versions that do not like self-assignment of boost::container::string
+  DNSName& operator=(const DNSName& a)
+  {
+    if (this != &a) {
+      d_storage = a.d_storage;
+    }
+    return *this;
+  }
+  DNSName(const DNSName& a) = default;
```

Comment:
I know this is a work-in-progress but be careful that declaring a copy assignment operator or a copy constructor will prevent the move operations from being generated.

Reply:
Yes. But so far this is the only "solution" I could think of.